exir/backend/utils.py

```diff
@@ -569,3 +572,81 @@ def __call__(self, node: torch.fx.Node, reason: str) -> None:
 
     def __str__(self) -> str:
         return f"WhyNoPartition: Node {self.node} was not partitioned because {self.reason}."
+
+
+class WhyNoPartitionReporter:
```

Comment:
cc @mcr229, @cccclai  - can you take a look at thi?

Reply:
@Erik-Lundell can you start a separate PR with this change so it's easier for discussion.

Reply:
can we make this as a util which can be shared by all the delegates? I wanted to do that when we first introduced it in xnnpack.